apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/OperationBasedModelGroupBuilder.kt

```diff
@@ -112,23 +114,44 @@ internal class OperationBasedModelGroupBuilder(
   /**
    * A grouping key for fragments
    */
-  private data class InlineFragmentKey(val typeCondition: String, val condition: BooleanExpression<BVariable>)
+  private data class InlineFragmentKey(val typeCondition: String, val condition: BooleanExpression<BTerm>)
 
-  private fun BooleanExpression<BVariable>.toName(): String = when (this) {
+  private fun BooleanExpression<BTerm>.toName(): String = when (this) {
     is BooleanExpression.True -> "True"
     is BooleanExpression.False -> "False"
     is BooleanExpression.And -> this.operands.joinToString("And") { it.toName() }
     is BooleanExpression.Or -> this.operands.joinToString("Or") { it.toName() }
     is BooleanExpression.Not -> "Not${this.operand.toName()}"
-    is BooleanExpression.Element -> this.value.name.capitalizeFirstLetter()
-    else -> error("")
+    is BooleanExpression.Element -> {
+      val bVariable = this.value as? BVariable ?: error("Unexpected term type in toName")
+      bVariable.name.capitalizeFirstLetter()
+    }
+    else -> error("Unexpected boolean expression type in toName")
   }
 
   private fun InlineFragmentKey.toName(): String = buildString {
     append(typeCondition.capitalizeFirstLetter())
     if (condition != BooleanExpression.True) {
-      append("If")
-      append(condition.toName())
+      // If at least one condition is a BLabel, it is enough to make a unique name, because the label is unique
+      val deferCondition: BLabel? = condition.firstElementOfType(BLabel::class)
```

Comment:
Maybe check that there is only one "label"? Else it's not balanced ⚖️  if there are multiple:

```graphql
{
  computer {
    ...computerDetails @defer("label1")
    ...computerDetails @defer("label2")
  }
}
```

I know this sample is contrieved and will most likely never happen but if it happens, the longer name shouldn't be too much of an issue (`ComputerDetailsLabel1OrLabel2`)

Reply:
Actually I don't understand 😅  This is only used for inline fragments, right?

Reply:
Right 👍 And inline fragments will always ever have one label. Works for me 👍